mti_nma/steps/avgshape/avgshape.py

```diff
@@ -26,29 +25,42 @@ def __init__(
         self,
         direct_upstream_tasks: Optional[List["Step"]] = [Shparam],
         filepath_columns=["AvgShapeFilePath", "AvgShapeFilePathStl"]
+
     ):
         super().__init__(
             direct_upstream_tasks=direct_upstream_tasks,
             filepath_columns=filepath_columns
         )
 
     @log_run_params
-    def run(self, mesh_density=5, **kwargs):
+    def run(self, mesh_density=5, sh_df=None, **kwargs):
         """
         This step uses the amplitudes of the spherical harmonic components
         of the nuclear shapes in the dataset to construct an average nuclear mesh.
+
+        Parameters
+        ----------
+        mesh_density: int (1-10)
+            Mesh density parameter used in Blender
+
+        sh_df: dataframe
+            dataframe containing results from running Shparam step
+            See the construction of the manifest in shparam.py for details
         """
 
-        # Get shparam manifest
-        shparam = Shparam()
-        manifest_filepaths_rel2abs(shparam)
-        df_sh = shparam.manifest.copy()
-        df_sh = df_sh.set_index("CellId", drop=True)
+        # if no dataframe is passed in, load manifest from previous step
+        if sh_df is None:
+            sh_df = pd.read_csv(
+                self.step_local_staging_dir.parent / "shparam" / "manifest.csv"
+            )
+
+        # fix filepaths and use cell id as dataframe index
+        sh_df = sh_df.set_index("CellId", drop=True)
 
         # Load sh coefficients of all samples in manifest
         df_coeffs = pd.DataFrame([])
-        for CellId in df_sh.index:
-            df_coeffs_path = df_sh["CoeffsFilePath"][CellId]
+        for CellId in sh_df.index:
```

Comment:
Rename sh_df by df_sh for consistency.

Reply:
Thanks for catching this! Everywhere else in the code actually uses the "_df" ordering, which is why I changed it here. Looks like df_coeffs is the one thats inconsistent so I will definitely fix that!

Reply:
Thanks for catching this! Everywhere else in the code actually uses the "_df" ordering, which is why I changed it here. Looks like df_coeffs is the one thats inconsistent so I will definitely fix that!

Reply:
Just added a commit changing to coeffs_df to fix the inconsistency